### PR TITLE
plymouth smooth transition and accountsservice support

### PR DESCRIPTION
**Pull request by xiangzhai**

### CODE REVIEW

**Comment by plfiorini** [comment]
Link to #174

**Comment by plfiorini** [comment]
This introduces a conundrum with the CI.
Builds will never pass unless we either build QtAccountsService before sddm or somebody does a package for Ubuntu 12.04...

**Comment by xiangzhai** [comment]
Hi @MingcongBai is there somebody does a QtAccountsService package for Ubuntu 12.04? thanks a lot!

**Comment by MingcongBai** [comment]
Why 12.04... sounds unresasonable...

**Comment by plfiorini** [comment]
@MingcongBai It is but Travis CI offers 12.04

**Comment by MingcongBai** [comment]
@plfiorini Alright... that's true. However it may be too much of a job to get Qt 5 into 12.04, just way too much time will be spent on this...

**Comment by xiangzhai** [comment]
Hi @plfiorini please review it again, thanks a lot!

**Comment by xiangzhai** [comment]
Hi @plfiorini I added plymouth smooth transition support, please review it, thanks a lot ;P

**Comment by tpgxyz** [comment]
Can you please prepare "clean" pull request to current sddm codebase ?

I'm asking for that because i'd like give it a testing, and picking up line by line from commits on your repo is taking me too much time. Thanks.

**Comment by Arthur2e5** [comment]
I will do some `rebase -i` to make the commits cleaner during the weekend.

**Comment by Arthur2e5** [comment]
Rebased, and partially reworded. This should look good enough, except that at a certain commit I chose the wrong README (theirs) to checkout and spotted it only in the final commit.

**Comment by Arthur2e5** [comment]
Uh-huh, what's that config-sddm thing? @xiangzhai

File: README.md [hunk 1]
```diff
@@ -1,21 +1,30 @@
 ## INTRODUCTION
 
 [![Build Status](https://travis-ci.org/sddm/sddm.svg?branch=master)](https://travis-ci.org/sddm/sddm)
```
Comment (Arthur2e5) [inline README.md]: Jeez, what happen here?

File: README.md [hunk 2]
```diff
@@ -1,21 +1,30 @@
...
-which can be used for authentication, suspend etc.
+SDDM is extremely themeable. We put no restrictions on the user interface design, it is completely up to the designer. We simply provide a few callbacks to the user interface which can be used for authentication, suspend etc. To further ease theme creation we provide some premade components like a textbox, a combox etc.
 
-To further ease theme creation we provide some premade components like a textbox,
-a combox etc.
+There are a few sample themes distributed with SDDM. They can be used as a starting point for new themes.
 
-There are a few sample themes distributed with SDDM.
-They can be used as a starting point for new themes.
+## LICENSING
```
Comment (Arthur2e5) [inline README.md]: The heck…

File: src/daemon/CMakeLists.txt [hunk 3]
```diff
@@ -4,28 +4,55 @@ include_directories(
...
-    XorgDisplayServer.cpp
-    Greeter.cpp
-    PowerManager.cpp
-    Seat.cpp
-    SeatManager.cpp
-    SignalHandler.cpp
-    SocketServer.cpp
-    VirtualTerminal.cpp
-)
+if(ENABLE_PLYMOUTH)
```
Comment (Arthur2e5) [inline src/daemon/CMakeLists.txt]: Jeeeez, is there a `+=` thing?

File: src/greeter/UserModel.cpp [hunk 4]
```diff
@@ -76,6 +80,9 @@ namespace SDDM {
             user->homeDir = QString::fromLocal8Bit(current_pw->pw_dir);
             user->uid = int(current_pw->pw_uid);
             user->gid = int(current_pw->pw_gid);
+#if HAVE_QTACCOUNTSSERVICE
+			QtAccountsService::UserAccount *userAccount = accountManager->findUserById(user->uid);
```
Comment (Arthur2e5) [inline src/greeter/UserModel.cpp]: Whoops, tabs, blame notepad++

**Comment by Arthur2e5** [comment]
Apparently I should add changelog entries too. Well, will you add them in the merge commit?

**Comment by xiangzhai** [comment]
在 2015年11月01日 14:43, Mingye Wang 写道:
>
> Uh-huh, what's that config-sddm thing? @xiangzhai 
> <https://github.com/xiangzhai>
>
you mean sddm-kcm? 
https://projects.kde.org/projects/kde/workspace/sddm-kcm/repository

**Comment by plfiorini** [comment]
This PR is a huge mess.
Please separate plymouth and qtaccountsservice code in two different forks.
I also suggest to fork current master and do the relevant changes in a few commits.

**Comment by Arthur2e5** [comment]
Let me try some cherry-picking. Just trying to keep the commit count for @xiangzhai…

**Comment by Arthur2e5** [comment]
By the way it's already based on some new enough commit. Since I often use `git pull --rebase=preserve some/repo someref` to merge PRs manually, I often only care about mergeability…

File: src/daemon/DisplayManager.cpp [hunk 5]
```diff
@@ -40,6 +44,13 @@ namespace SDDM {
         QDBusConnection connection = (daemonApp->testing()) ? QDBusConnection::sessionBus() : QDBusConnection::systemBus();
         connection.registerService(DISPLAYMANAGER_SERVICE);
         connection.registerObject(DISPLAYMANAGER_PATH, this);
+#if HAVE_PLYMOUTH 
+        Plymouth::quitWithTransition();
+#endif
+    }
+
+    DisplayManager::~DisplayManager()
```
Comment (Arthur2e5) [inline src/daemon/DisplayManager.cpp]: Wait, why did we add an empty destructor…

Comment (xiangzhai) [inline src/daemon/DisplayManager.cpp]: yes, it is useless, just remove it ;-)

**Comment by tpgxyz** [comment]
Hi,
can you guys prepare a clean PR ?

**Comment by Arthur2e5** [comment]
Almost forgot this. Be patient, I will clean those up in a week…

**Comment by beojan** [comment]
Are there any updates on this pull request?

**Comment by MingcongBai** [comment]
@Arthur2e5 Will you please...

**Comment by xiangzhai** [comment]
Hi @beojan Cleanup now ;-) please use ```sudo systemctl disable sddm``` and ```sudo systemctl enable sddm-plymouth```

**Comment by xiangzhai** [comment]
Hi @tpgxyz please review it, thanks a lot!

**Comment by Arthur2e5** [comment]
#525 updated.

File: src/daemon/Plymouth.cpp [hunk 6]
```diff
@@ -1,116 +0,0 @@
-/***************************************************************************
```
Comment (Arthur2e5) [inline src/daemon/Plymouth.cpp]: ?

Comment (xiangzhai) [inline src/daemon/Plymouth.cpp]: yes

**Comment by beojan** [comment]
Currently SDDM is failing to start with themes that use Plasma components (e.g. Plasma 5's breeze theme).

I don't know if it is a bug in the rest of sddm, or in this pull request, but since I am using this branch I think it's best to report it here first.

**Comment by xiangzhai** [comment]
hi @beojan please switch to tty2, then sudo systemctl restart sddm or sddm-plymouth

**Comment by beojan** [comment]
Yes, I've tried restarting sddm. I found that it doesn't fail if the Plasma theme for the sddm user (i.e. that set in `/var/lib/sddm/.config/plasmarc`) is air, while it does fail with, for example, the default theme (i.e. breeze).

**Comment by xiangzhai** [comment]
@beojan with breeze theme, ***restart sddm*** still failed to show the greeter?

**Comment by beojan** [comment]
Indeed. `journalctl`  shoes that SDDM has segfaulted.

**Comment by beojan** [comment]
Actually, it turns out the issue was that the permissions were set incorrectly on the Plasma theme's `metadata.desktop` file, so SDDM couldn't read it.

**Comment by xiangzhai** [comment]
@beojan it is better to report the bug https://github.com/sddm/sddm/issues/new

**Comment by beojan** [comment]
Is this going to be merged soon? If not (or even if so), can you update your branch to match the latest updates?

**Comment by xiangzhai** [comment]
@beojan updated ;-)

**Comment by plfiorini** [comment]
I'm sorry but this is unmergeable. Please make a clean PR.

**Pull request closed without merging**